src/nft/pages/collection/index.tsx

```diff
@@ -1,5 +1,35 @@
-export const Collection = () => {
-  return <div>Collection Page</div>
+import { useWindowSize } from 'hooks/useWindowSize'
+import { GenieCollection } from 'nft/types'
+import { useEffect, useMemo } from 'react'
+import { useQuery } from 'react-query'
+import { useLocation, useParams } from 'react-router-dom'
+
+import { useIsMobile } from '../../hooks/useIsMobile'
+import { CollectionStatsFetcher } from '../../queries'
+import { CollectionDesktop } from './CollectionDesktop'
+import { CollectionMobile } from './CollectionMobile'
+
+const Collection = () => {
+  const { contractAddress } = useParams()
+
+  const isMobile = useIsMobile((state) => state.isMobile)
+  const setMobileWidth = useIsMobile((state) => state.setMobileWidth)
+  const location = useLocation()
+  const isActivityToggled = useMemo(() => location.pathname.includes('/activity'), [location])
+
+  const { data: collectionStats } = useQuery(['collectionStats', contractAddress], () =>
+    CollectionStatsFetcher(contractAddress as string)
+  )
+
+  const { width } = useWindowSize()
+  useEffect(() => {
+    setMobileWidth(width ?? 0)
+  }, [width, setMobileWidth])
+
+  if (isMobile)
+    return <CollectionMobile {...{ collectionStats: collectionStats ?? ({} as GenieCollection), isActivityToggled }} />
```

Comment:
`{}` does not satisfy `GenieCollection`, so you shouldn't coerce it, as it may lead to problems in future refactors. Instead, can you have the `CollectionStats` check for `undefined` and return `null` in that case?